pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+                 verbose=False):
+        """
+        The constructor downloads the schema if the 'schema_dir' does not exist
+        and installs it into the directory defined by the 'schema_dir'
+        attribute.
+
+        If the schema is already installed it simply sets the property values so
+        that the schema can be used.
+
+        If the schema was previously downloaded but not expanded, it is
+        automatically expanded (i.e. the schema/mof subdirectory is created
+        and the mof files expanded into this directory.). This allows only
+        the single file 'schema_mof_filename' to be persisted and the expansion
+        to occur on usage which saves downloading from the DMTF repeatedly and
+        saves disk space.
+
+        Parameters:
+
+          schema_dir (:term:`string`):
+            Directory into which the DMTF schema is installed or will be
+            installed.
+
+          schema_ver  (tuple of 3 integers (x, y, z):
+            Represents the DMTF the schema version in the format x.yy.z where
+                x is the major version
+                y is the minor version
+                z is the revisions
+            This MUST represent a DMTF schema that is available from the
+            DMTF web site.
+
+          use_experimental (:class:`py:bool`):
+            If True the expermintal version of the defined schema is
+            installed.
+
+            If False (the default) the Final released version of the DMTF
+            is installed.
+
+          verbose (:class:`py:bool`):
+            If 'True' status messages are output to the console
+
+        Raises:
+            ValueError if the schema cannon be retrieved from the DMTF web site.
+            ValueError or if the 'schema_dir' differs from the version of the
+            schema currently set into the 'schema_dir'
+
+        """
+        if not isinstance(schema_ver, tuple):
+            raise ValueError('schema_ver must be tuple not %s' % schema_ver)
+        if len(schema_ver) != 3:
+            raise ValueError('DMTF Schema must have 3 integer components, '
+                             '(major, minor, rev) not %s' % schema_ver)
+        schema_type = 'Experimental' if use_experimental else 'Final'
+        self._schema_ver = schema_ver
+        self._schema_dir = schema_dir
+
+        mof_dir = 'mof%s%s' % (schema_type, self.schema_ver_str)
+        self._schema_mof_dir = os.path.join(self._schema_dir, mof_dir)
+
+        self._dmtf_schema_ver = 'cim_schema_%s' % (self.schema_ver_str)
+        self._mof_zip_bn = '%s%s-MOFs.zip' % (self._dmtf_schema_ver,
+                                              schema_type)
+        self._mof_zip_url = \
+            'http://www.dmtf.org/standards/cim/cim_schema_v%s%s%s/%s' % \
+            (schema_ver[0], schema_ver[1], schema_ver[2], self._mof_zip_bn)
+        self._schema_mof_bld_name = self._dmtf_schema_ver + '.mof'
```

Comment:
Do we really need to store intermediate strings such as `_mof_zip_bn` and `_schema_mof_bld_name` in the object?

Reply:
No we don't.  Made them local.  I need to review once more to be sure I have correctly moved everything to local I wanted to but generally

DONE